utils/crypto/bls/signer/localsigner/bls_test.go

```diff
@@ -24,79 +24,89 @@ func AggregateAndVerify(publicKeys []*bls.PublicKey, signatures []*bls.Signature
 	return bls.Verify(aggPK, aggSig, message), nil
 }
 
-func NewKeyPair(require *require.Assertions) (*LocalSigner, *bls.PublicKey) {
+func NewSigner(require *require.Assertions) *LocalSigner {
 	sk, err := New()
 	require.NoError(err)
-	pk := sk.PublicKey()
-	return sk, pk
+	return sk
+}
+
+func Map[T any, U any](arr []T, f func(T) U) []U {
+	result := make([]U, len(arr))
+	for i, val := range arr {
+		result[i] = f(val)
+	}
+	return result
 }
```

Comment:
I think we should try to be in the habit of implementing helpers in better places. So if we actually want to add this it should probably be in a place that can be shared.

That being said, I don't really think map/filter are very standard for golang (they are more natural in a more functional language).

They were actually declined to be added to the stdlib here: https://github.com/golang/go/discussions/47203 (That being said, it wasn't a final decision. It was left as a potential future addition).